### ISSUE TITLE
Update TreeDataProvider onDidChangeTreeData event type

### ISSUE THREAD

**Issue by RMacfarlane** [microsoft/vscode#96932]
Issue Type: <b>Bug</b>

With the change from https://github.com/microsoft/vscode/issues/93875, the data fired from an event emitter is no longer optional.

For extensions using the `TreeDataProvider` API, firing an emitter for `onDidChangeTreeData` without any arguments is now a compilation error. 

```
/**
 * An optional event to signal that an element or root has changed.
 * This will trigger the view to update the changed element/root and its children recursively (if shown).
 * To signal that root has changed, do not pass any argument or pass `undefined` or `null`.
 */
onDidChangeTreeData?: Event<T | undefined | null>;
```

We should add `void` to the type list here.

VS Code version: Code - Insiders 1.45.0-insider (fd5d9a30259bb2bebab3e62dfd8315c2e1b27744, 2020-05-04T09:21:17.770Z)
OS version: Darwin x64 18.7.0

**Comment by wapenshaw** [comment]
Hi, has this been implemented? I am on VSCode 1.45.1 and I still get this error when trying to compile.

![image](https://user-images.githubusercontent.com/739546/83967185-c587ab00-a8ae-11ea-899a-f0e9da53ac52.png)

**Comment by RMacfarlane** [comment]
Yes, but the EventEmitter still needs to be defined with type `void`, `EventEmitter<SpaceItem | undefined | void>`
